src/HotChocolate/Core/src/Types/Types/Descriptors/Conventions/XmlDocumentationProvider.cs

```diff
@@ -31,28 +33,75 @@ public class XmlDocumentationProvider
             {
                 throw new ArgumentNullException(nameof(fileResolver));
             }
+
             _fileResolver = fileResolver;
         }
 
-        public string GetSummary(Type type) =>
-            GetSummary((MemberInfo)type);
+        public string GetDescription(Type type) =>
+            GetDescription((MemberInfo)type);
 
-        public string GetSummary(MemberInfo member)
+        public string GetDescription(MemberInfo member)
         {
             var assemblyName = member.Module.Assembly.GetName();
             var element = GetMemberElement(member);
-            return RemoveLineBreakWhiteSpaces(
-                GetText(element?.Element(_summaryElementName)));
+
+            XElement summary = element?.Element(_summaryElementName);
+
+            XElement returns = element?.Element(_returnsElementName);
+
+            var exceptions = element?.Elements(_exceptionElementName).ToList();
+
+            var description = ComposeMemberDescription(summary, returns, exceptions);
+            var normalizedDescription = RemoveLineBreakWhiteSpaces(description);
+
+            return normalizedDescription;
         }
 
-        public string GetSummary(ParameterInfo parameter)
+        public string GetDescription(ParameterInfo parameter)
         {
             var assemblyName = parameter.Member.Module.Assembly.GetName();
             var element = GetParameterElement(parameter);
             return RemoveLineBreakWhiteSpaces(
                 GetText(element));
         }
 
+        private string ComposeMemberDescription(XElement summary, XElement returns, List<XElement> exceptions)
+        {
+            var builder = new StringBuilder();
+            if (!string.IsNullOrEmpty(summary?.Value))
+            {
+                builder.Append(GetText(summary));
+            }
+
+            if (!string.IsNullOrEmpty(returns?.Value))
+            {
+                builder.Append(Environment.NewLine);
+                builder.AppendLine($"**Returns:** {GetText(returns)}");
+            }
+
+            if (exceptions != null)
+            {
+                var errorNumber = 1;
+                var exceptionStrings =
+                    (from ex in exceptions
+                    let code = ex.Attribute("code")
+                    where !string.IsNullOrEmpty(ex.Value) && code != null
+                    select $"{errorNumber++}. {code.Value}: {GetText(ex)}").ToList();
```

Comment:
Also get rid f this instance of list since we know that if we find no elements our enumerable will be null.

Reply:
Thinking about it my suggestion is not true :) since we are filtering out code.